### PR TITLE
infra: Use GitHub source for tox tests

### PR DESCRIPTION
**Pull request by speller26**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -78,3 +83,10 @@ basepython = python3
 skip_install = true
 commands =
     /bin/sh -c 'tar -czvf build_files.tar.gz build/'
+
+[test-deps]
+deps =
+# If you need to test on a certain branch, add @<branch-name> after .git
+    git+https://github.com/aws/amazon-braket-schemas-python.git
+    git+https://github.com/aws/amazon-braket-default-simulator-python.git
```
Comment (kshitijc) [inline tox.ini:91]: Is the dependency on default simulator needed?

Comment (speller26) [reply tox.ini:91]: Nope, good catch

Comment (speller26) [inline tox.ini:91]:
```suggestion
    git+https://github.com/aws/amazon-braket-schemas-python.git
    git+https://github.com/aws/amazon-braket-sdk-python.git
```

File: tox.ini [hunk 2]
```diff
@@ -78,3 +83,10 @@ basepython = python3
 skip_install = true
 commands =
     /bin/sh -c 'tar -czvf build_files.tar.gz build/'
+
+[test-deps]
+deps =
+# If you need to test on a certain branch, add @<branch-name> after .git
+    git+https://github.com/aws/amazon-braket-schemas-python.git
+    git+https://github.com/aws/amazon-braket-sdk-python.git
+    git+https://github.com/aws/amazon-braket-sdk-python.git
```
Comment (kshitijc) [inline tox.ini:92]: Duplicated lines

Comment (speller26) [reply tox.ini:92]: Ugh, should've looked at context lines.

Comment (speller26) [reply tox.ini:92]:
```suggestion
    git+https://github.com/aws/amazon-braket-sdk-python.git
```

Comment (speller26) [inline tox.ini:92]:
```suggestion
    git+https://github.com/aws/amazon-braket-sdk-python.git
```

**Review by kshitijc — APPROVED** [review]

**Pull request merged**